src/math/interpolations/newton.lisp

```diff
@@ -0,0 +1,85 @@
+(in-package :lbge.math)
+
+(defclass newton-interp ()
+  ((divdiffs :initarg :divdiffs :accessor divdiffs)
+   (xs :initarg :xs :accessor xs)
+   (poly :initarg :poly :accessor poly)))
+
+(defmethod degree ((interp newton-interp))
+  (length (xs interp)))
+
+(defun div-diff (xs row start order)
+  (let ((y0 (nth (1+ start) row))
+        (y1 (nth start row))
+        (x0 (nth (+ start order) xs))
+        (x1 (nth start xs)))
+      (/ (- y0 y1)
+         (- x0 x1))))
+
+(defun calc-div-diffs (xs ys)
+  (let ((diffs (list ys))
+        (len (length ys)))
+     (loop for o from 1 below len
+      do (append-to diffs
+                    (loop for n from 0 below (- len o)
+                          collect (div-diff xs (nth (1- o) diffs) n o))))
+     diffs))
+
+(defun calc-basis-polynomial (interp order)
+  (let ((p (make-polynomial 1)))
+       (loop for x in (subseq (xs interp) 0 order)
+             do (setf p (mul p (make-polynomial (- x) 1))))
+      p))
+
+(defun calc-newton-polynomial (interp)
+  (let ((p (make-polynomial 0)))
+      (loop for i from 0 below (degree interp)
+            do (setf p (add p
+                            (mul (calc-basis-polynomial interp i)
+                                 (first (nth i (divdiffs interp)))))))
+      p))
+
+(defun make-newton-raw (coords)
+  (assert (>= (length coords) 3)
+          nil
+          "There must be at least 3 points to interpolate")
+  (let ((xs '())
```

Comment:
'() is literal constant data.
Assigning it to variables in this form means that xs and ys will remain constant and will not be modified, and this is not true.
Thus, the correct choice for xs and ys initialization would be `(list)`.